packages/react-data-grid/src/HeaderCell.tsx

```diff
@@ -1,168 +1,118 @@
-      if (!touch) return;
-      window.removeEventListener('touchmove', onTouchMove);
-      window.removeEventListener('touchend', onTouchEnd);
-    };
-
-    window.addEventListener('touchmove', onTouchMove);
-    window.addEventListener('touchend', onTouchEnd);
-  };
-
-  private onResize(x: number) {
-    const width = this.getWidthFromMouseEvent(x);
-    if (width > 0) {
-      this.props.onResize(this.props.column, width);
-    }
-  }
-
-  private getWidthFromMouseEvent(x: number): number {
-    return x - this.cell.current!.getBoundingClientRect().left;
-  }
-
-  getCell() {
-    const { height, column, rowType, allRowsSelected, onAllRowsSelectionChange } = this.props;
-    const renderer = this.props.renderer || SimpleCellRenderer;
+export default function HeaderCell<R, K extends keyof R>({
+  height,
+  column,
+  allRowsSelected,
+  onAllRowsSelectionChange,
+  scrollLeft,
+  ...props
+}: HeaderCellProps<R, K>) {
+  function getCell() {
+    const renderer = column.headerRenderer || SimpleCellRenderer;
     if (isElement(renderer)) {
+      // TODO: is this needed?
       // if it is a string, it's an HTML element, and column is not a valid property, so only pass height
       if (typeof renderer.type === 'string') {
-        return React.cloneElement(renderer, { height });
+        return cloneElement(renderer, { height });
       }
-      return React.cloneElement(renderer, { column, height });
+      return cloneElement(renderer, { column, height });
     }
-    return React.createElement(renderer, { column, rowType, allRowsSelected, onAllRowsSelectionChange });
+    return createElement(renderer, { column, allRowsSelected, onAllRowsSelectionChange });
   }
 
-  setScrollLeft(scrollLeft: number) {
-    const node = this.cell.current;
-    if (node) {
-      node.style.transform = `translateX(${scrollLeft}px)`;
-    }
-  }
+  let cell = getCell();
 
-  removeScroll() {
-    const node = this.cell.current;
-    if (node) {
-      node.style.transform = 'none';
-    }
+  if (column.sortable) {
+    cell = (
+      <SortableHeaderCell
+        column={column}
+        onSort={props.onSort}
+        sortColumn={props.sortColumn}
+        sortDirection={props.sortDirection}
+      >
+        {cell}
+      </SortableHeaderCell>
+    );
   }
 
-  render() {
-    const { column, rowType } = this.props;
-    const colIsFrozen = isFrozen(column);
+  const colIsFrozen = isFrozen(column);
+  const className = classNames('rdg-cell', {
+    'rdg-cell-frozen': colIsFrozen,
+    'rdg-cell-frozen-last': colIsFrozen && column.idx === props.lastFrozenColumnIndex
+  }, column.cellClass);
+  const style: React.CSSProperties = {
+    width: column.width,
+    left: column.left
+  };
 
-    const className = classNames('rdg-cell', {
-      'rdg-cell-frozen': colIsFrozen,
-      'rdg-cell-frozen-last': colIsFrozen && column.idx === this.props.lastFrozenColumnIndex
-    }, this.props.className, column.cellClass);
+  if (typeof scrollLeft === 'number') {
```

Comment:
Do we not need to check if the column is frozen here as well?
We seem to do it in `FilterRow.tsx` and `Row.tsx`.

Reply:
We need to check it as we set `scrollLeft` on frozen columns only. `FilterRow` does not use `HeaderCell` component or the `Row` component, it directly renders the filter component. We can probably do some clean up and extract this common logic somewhere